config/v1/types_infrastructure.go

```diff
@@ -25,7 +25,9 @@ type InfrastructureSpec struct {
 }
 
 type InfrastructureStatus struct {
-	// type
+	// cloudProvider is the IaaS provider that is running the cluster. The value
+	// should be a valid kubelet --cloud-provider value.
```

Comment:
@openshift/api-reviewers We should encourage documentation in `config.openshift.io` to stand alone and not depend on other binaries for their docs.  

@ironcladlou in this case, this should list out the allowed values.  Then it can be enforced in validation (something like https://github.com/openshift/origin/pull/21679 or whatever we end up choosing).

Also, squash

Reply:
Done